dagobah/core/delegator.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+from ..backend.base import BaseBackend
+
+logger = logging.getLogger('dagobah')
+
+class CommitDelegator(object):
+    """
+    The CommitDelegator is a class designed to handle any and all commits
+    to the database.
+
+    The delegator takes in commit requests from various objects, and decides
+    if they should actually be committed, and if so, manage any special
+    processing.
+    """
+
+    def __init__(self, backend=BaseBackend()):
```

Comment:
don't think you want to provide a default backend here, we already do this in the Dagobah class constructor (can't really remember why), but failing to provide one to this class is probably a programming error